### ISSUE TITLE
[autoscaler] Api instead of CLI to interact with cluster.

### ISSUE THREAD

**Issue by jieru-hu** [ray-project/ray#8036]
### Describe your feature request
It would be great if autoscaler provides an API in addition to the CLI. For example, to spin up a cluster instead of doing "ray up cluster.yaml", we can do "ray.up(cluster_config: Dict[str, Any])". Having a API would provide a good experience interacting autoscaler programmatically.

**Comment by richardliaw** [comment]
You can currently do this via `ray.autoscaler.commands`: 

See: `ray/python/ray/autoscaler/commands.py`. However, we can probably improve this moving forward.

**Comment by anabranch** [comment]
@richardliaw is this a supported API or is this subject to change?

**Comment by richardliaw** [comment]
It's not a public-facing API, but there's no plans to change it.

**Comment by omry** [comment]
> You can currently do this via `ray.autoscaler.commands`:
> 
> See: `ray/python/ray/autoscaler/commands.py`. However, we can probably improve this moving forward.

One thing I would like to see there is a way to not pass in the config directly without having to serialize it to a file first.

**Comment by anabranch** [comment]
@omry to confirm, that's the request in #8036 right?

**Comment by ericl** [comment]
I don't think we want to maintain this as a public API. You can use subprocess call to invoke the CLI instead.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
